doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst

```diff
@@ -182,8 +182,43 @@ To bring a stalled Target node to idle state, use the ``mesh models dfu cli canc
 
 .. _bluetooth_mesh_dfu_eval_md:
 
-Composing the firmware metadata
-*******************************
+Generating the firmware metadata
+********************************
+
+There are two ways to generate the required DFU metadata:
+
+  * Automated generation using the DFU metadata extraction script integrated in the |NCS| build system.
+  * Manual generation by using shell commands.
+
+Using the DFU metadata extraction script is the most efficient way of generating the required DFU metadata.
+
+Automated metadata generation
+=============================
+
+By enabling the :kconfig:option:`CONFIG_BT_MESH_DFU_METADATA_ON_BUILD` option in the application, the metadata will be automatically parsed from the ``.elf`` and ``.config`` files.
+The parsed data is stored in the :file:`ble_mesh_metadata.json` file.
+The file is placed in the :file:`dfu_application.zip` archive in the build folder of the application.
+Additionally, the metadata string required by the ``mesh models dfu slot add`` command will be printed in the command line window when the application is built::
+
+  Bluetooth Mesh Composition metadata generated:
+    Encoded metadata: 020000000100000094cf24017c26f3710100
+    Full metadata written to: APPLICATION_FOLDER\build\zephyr\dfu_application.zip
+
+.. note::
+   Currently this script only supports applications that contain a single composition data structure, meaning that applications containing multiple composition data structures must rely on manual generation of the metadata string(s).
+   Additionally, the script is hardcoded to produce a metadata string where the firmware is targeted for the application core.
+
+In addition to printing the metadata string when the application is built, a separate west command can be utilized to print the metadata to the console, given that it is already generated by the build system.
```

Comment:
The last part of the sentence tells the same as the first one. I'd suggest to remove the first part of the sentence:
```suggestion
A separate west command can be utilized to print the metadata to the console, given that it is already generated by the build system.
```